tools/Config.mk

```diff
@@ -555,26 +555,34 @@ endef
 
 # ARCHxxx means the predefined setting(either toolchain, arch, or system specific)
 
-ARCHDEFINES += ${shell $(DEFINE) "$(CC)" __NuttX__}
+ifndef DEFINE_PREFIX
+export DEFINE_PREFIX:= $(subst __DEF__,,${shell $(DEFINE) "$(CC)" "__DEF__"})
```

Comment:
Should have space before `:=`?

Reply:
I think that it is good to have

Reply:
let's modify DEFINE/INCDIR directly and replace all places?

Reply:
> let's modify DEFINE/INCDIR directly and replace all places?

This has been done, the apps repository will be updated after this PR is merged.
The original way still works